x-pack/solutions/observability/plugins/slo/public/embeddable/slo/alerts/slo_alerts_embeddable_factory.tsx

```diff
@@ -72,52 +68,68 @@ export function getAlertsEmbeddableFactory({
         }
       }
 
-      const titleManager = initializeTitleManager(state);
+      const titleManager = initializeTitleManager(initialState.rawState);
+      const sloAlertsStateManager = initializeStateManager<EmbeddableSloProps>(
+        initialState.rawState,
+        {
+          slos: [],
+          showAllGroupByInstances: false,
+        }
+      );
       const defaultTitle$ = new BehaviorSubject<string | undefined>(getAlertsPanelTitle());
-      const slos$ = new BehaviorSubject(state.slos);
-      const showAllGroupByInstances$ = new BehaviorSubject(state.showAllGroupByInstances);
       const reload$ = new Subject<FetchContext>();
-      const api = buildApi(
-        {
-          ...titleManager.api,
-          defaultTitle$,
-          getTypeDisplayName: () =>
-            i18n.translate('xpack.slo.editSloAlertswEmbeddable.typeDisplayName', {
-              defaultMessage: 'configuration',
-            }),
-          isEditingEnabled: () => true,
-          onEdit: async () => {
-            onEdit();
-          },
-          serializeState: () => {
-            return {
-              rawState: {
-                ...titleManager.serialize(),
-                slos: slos$.getValue(),
-                showAllGroupByInstances: showAllGroupByInstances$.getValue(),
-              },
-            };
-          },
-          getSloAlertsConfig: () => {
-            return {
-              slos: slos$.getValue(),
-              showAllGroupByInstances: showAllGroupByInstances$.getValue(),
-            };
-          },
-          updateSloAlertsConfig: (update) => {
-            slos$.next(update.slos);
-            showAllGroupByInstances$.next(update.showAllGroupByInstances);
+
+      function serializeState() {
+        return {
+          rawState: {
+            ...titleManager.getLatestState(),
+            ...sloAlertsStateManager.getLatestState(),
           },
+        };
+      }
+
+      const unsavedChangesApi = initializeUnsavedChanges({
+        uuid,
+        parentApi,
+        serializeState,
+        anyStateChange$: merge(titleManager.anyStateChange$, sloAlertsStateManager.anyStateChange$),
+        getComparators: () => ({
+          ...titleComparators,
+          slos: 'referenceEquality',
+          showAllGroupByInstances: 'referenceEquality',
+        }),
+        onReset: (lastSaved) => {
+          titleManager.reinitializeState(lastSaved?.rawState);
+          sloAlertsStateManager.reinitializeState(lastSaved?.rawState);
         },
-        {
-          slos: [slos$, (value) => slos$.next(value)],
-          showAllGroupByInstances: [
-            showAllGroupByInstances$,
-            (value) => showAllGroupByInstances$.next(value),
-          ],
-          ...titleManager.comparators,
-        }
-      );
+      });
+
+      const api = finalizeApi({
+        ...titleManager.api,
+        ...unsavedChangesApi,
+        defaultTitle$,
+        hideTitle$: titleManager.api.hidePanelTitles$,
```

Comment:
What is `hideTitle$` and `setHideTitle` why are they added to the api? Title apis should be provided by `...titleManager.api,`

Reply:
You should not longer need these. https://github.com/elastic/kibana/pull/218369 has resolved the issue and now provides `hideTitle$` and `setHideTitle` in the `titleManager.api`

Reply:
Awesome! Thanks for updating the titleManager.